aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLCommandProcessor.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.storage.sqlite;
+
+import android.database.Cursor;
+import android.database.SQLException;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteStatement;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.logging.Logger;
+
+import java.util.List;
+import java.util.Objects;
+
+final class SQLCommandProcessor {
```

Comment:
Can you a class doco describing the single responsibility for the component? We have a number of various SQL gizmos now & it would help us to keep track in the future.